File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/aop/bigimg/picasso/PicassoHook.java
package com.didichuxing.doraemonkit.aop.bigimg.picasso;

import android.net.Uri;


import com.squareup.picasso.Transformation;

import java.util.ArrayList;
import java.util.List;

/**
 * ================================================
 * 作    者：jint（金台）
 * 版    本：1.0
 * 创建日期：2020/3/23-13:45
 * 描    述：
 * 修订历史：
 * ================================================
 */
public class PicassoHook {

    /**
     * 注入到com.squareup.picasso.Request 构造方法中
     */
    public static List<Transformation> proxy(Uri uri, List<Transformation> transformations) {
        try {
            if (transformations == null) {
                transformations = new ArrayList<>();
                transformations.add(new DokitPicassoTransformation(uri));
            } else {
                transformations.add(new DokitPicassoTransformation(uri));
            }
            return transformations;
        } catch (Exception e) {
            e.printStackTrace();
        }

        return null;

    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/aop/bigimg/picasso/PicassoHook.kt
package com.didichuxing.doraemonkit.aop.bigimg.picasso

import com.blankj.utilcode.util.ReflectUtils
import com.squareup.picasso.Request
import java.util.*

/**
 * ================================================
 * 作    者：jint（金台）
 * 版    本：1.0
 * 创建日期：2020/3/23-13:45
 * 描    述：
 * 修订历史：
 * ================================================
 */
public object PicassoHook {
    /**
     * 注入到com.squareup.picasso.Request 构造方法中
     */
    @JvmStatic
    public fun proxy(request: Any?) {
        try {
            if (request is Request) {
                val requestObj = request
                var transformations = requestObj.transformations
                if (transformations == null) {
                    transformations = ArrayList()
                    transformations.add(DokitPicassoTransformation(requestObj.uri, requestObj.resourceId))
                } else {
                    transformations.clear()
                    transformations.add(DokitPicassoTransformation(requestObj.uri, requestObj.resourceId))
                }
                ReflectUtils.reflect(request).field("transformations", transformations)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}